### PR TITLE
feat: release schedule

### PR DESCRIPTION
**Pull request by clavin**
#### Description of Change

Adds a schedule that shows key dates about major Electron releases:

* Stable, beta, and alpha release dates
* End-of-life dates
* Chromium & Node.js versions

Screenshot 2025-11-04 at 11 27 13 PM

This information matches the [manual table in the docs](https://www.electronjs.org/docs/latest/tutorial/electron-timelines#timeline).

#### Checklist
- [x] This PR was not created with AI. (PRs created mainly with AI will be closed. They waste our team's time. We ban repeat offenders.)

### CODE REVIEW

File: app/routes/history.tsx [hunk 1]
```diff
@@ -206,7 +206,7 @@ export default function ReleaseHistory() {
       </div>
 
       <div className="bg-white dark:bg-gray-800 rounded-xl shadow-sm border border-gray-200 dark:border-gray-700 overflow-hidden mb-6">
-        <div className="p-4 border-b border-gray-200 dark:border-gray-700 bg-gray-50 dark:bg-gray-900/50 flex flex-wrap items-center gap-6">
+        <div className="p-4 border-gray-200 dark:border-gray-700 bg-gray-50 dark:bg-gray-900/50 flex flex-wrap items-center gap-6">
```
Comment (clavin) [inline app/routes/history.tsx]: Drive-by touchup: this bottom border was redundant. (Looking again, I think the colors might be too? But at least it doesn't show now.)

**Comment by clavin** [comment]
P.S. I added stripes to the nightly & prerelease rows because (1) I think they're cool and (2) they help disambiguate the green and yellow for deuteranopic color blindness (and verified this with my wonderful deuteranopic partner).

File: app/data/release-schedule.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+type AbsoluteMajorReleaseSchedule = Omit<MajorReleaseSchedule, 'status'>;
+
+interface MajorReleaseGroup {
+  major: number;
+  releases: ElectronRelease[];
+  firstStable?: ElectronRelease; // Only used for Chromium milestone extraction
+}
+
+const SCHEDULE_OVERRIDES: Map<string, Partial<AbsoluteMajorReleaseSchedule>> = new Map([
```
Comment (dsanders11) [inline app/data/release-schedule.ts]: I think we need more adjustments here, there's more discrepancies with the dates at https://www.electronjs.org/docs/latest/tutorial/electron-timelines for the historical dates, particularly betas.

When I was attempting this I was [pulling those historical dates from the releases data](https://github.com/electron/release-status/compare/44c966501cc724d2909caec55ead9ef4bbde47e5...versions-json-refactor#diff-8190b0aaf866b637d1461f25e6c188f27afce6da67cac9f2bb9362790ce02110R92-R103). Maybe you want to consider doing that? The actual release dates in that file should take precedence over the calculated schedule dates, and would require fewer overrides.

File: app/data/release-schedule.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        const prevStable = new Date(prevSchedule.stableDate + 'T00:00:00');
+        prevStable.setDate(prevStable.getDate() + 2);
+        alphaDate = prevStable.toISOString().split('T')[0];
+      }
+
+      const group = majorGroups.get(major)!;
+      const latestRelease = group.releases[0];
+
+      const entry: AbsoluteMajorReleaseSchedule = {
+        version: `${major}.0.0`,
```
Comment (dsanders11) [inline app/data/release-schedule.ts]: For discussion: do we really want the `.0.0` suffix here, or just the major? Since the Node.js version shown is for the latest release in a major line, it's a bit misleading to show that Node.js version and the `.0.0` version number, since that might cause confusion.

I'd be in favor of dropping the suffix and just going with the major numbers, personally. I don't think the static suffix on all of them adds any value (I know this is how we have it in the Markdown at the moment).

**Review by dsanders11 — CHANGES_REQUESTED** [review]
Absolutely love this.

I've wanted this data in programmatic form for years now (see #18 for an earlier attempt at just JSON output, which I later refactored on the [`versions-json-refactor` branch](https://github.com/electron/release-status/tree/versions-json-refactor). Non-blocking for this PR, but I'd want to also expose this data in the future via a `/schedule.json` endpoint - that's the last piece of the puzzle for putting proper dates into our Releases WG project board stable prep items. If we expose it as JSON I should be able to update the automation around templating to grab the right dates and fill them in for us.

I've left some inline comments, but a big meta one: tests? Some inspiration from [the tests I was fleshing out on my branch](https://github.com/electron/release-status/blob/5ec2bd9ebdab85f293eda60462012f484b1cd1b6/test/data.spec.js#L65-L77). Namely we should test the edge cases (when alphas started, etc) and a snapshot in time of correct data would be nice so that we can change this code without risking missing subtle changes to the dates.

File: app/data/release-schedule.ts (hunk 3 above)
Comment (dsanders11) [reply app/data/release-schedule.ts]: > The Node.js version shown is for the `*.0.0` release—unless there are no stable releases yet, in which case it uses the latest pre-release for that line.

Ah, I misread the code, you're right. With that in mind, I'm fine leaving it with the `.0.0` suffix for now.

Alternatively, we could make the suffix `.x.y` which would leave the visual anchor, but avoid any confusion about what info applies to the whole major line versus just the `.0.0` version.

**Review by dsanders11 — APPROVED** [review]

**Comment by clavin** [comment]
Just realized that in [the doc](https://www.electronjs.org/docs/latest/tutorial/electron-timelines) we've been listing beta releases on Wednesdays when we internally plan around them releasing on Tuesdays. 💀 I'm going to retroactively correct the old dates (v27-v39) back to the targeted Tuesday (thus for the future too) here and leave a note in the doc about the correction.

**Comment by clavin** [comment]
@dsanders11 Sorry -- need a fresh review after the merge commit 😄
